jedeschule/spiders/baden_württemberg.py

```diff
@@ -0,0 +1,98 @@
+    def parse(self, response):
+        links_url = 'https://lobw.kultus-bw.de/didsuche/DienststellenSucheWebService.asmx/SearchDienststellen'
+        timestamp = str(int(time.time())) 
+        body = {"command":"QUICKSEARCH",
+                    "data":{
+                        "dscSearch":"",
+                        "dscPlz":"",
+                        "dscOrt":"",
+                        "dscDienststellenname":"",
+                        "dscSchulartenSelected":"",
+                        "dscSchulstatusSelected":"",
+                        "dscSchulaufsichtSelected":"",
+                        "dscOrtSelected":"",
+                        "dscEntfernung":"",
+                        "dscAusbildungsSchulenSelected":"",
+                        "dscAusbildungsSchulenSelectedSart":"",
+                        "dscPageNumber":"1",
+                        "dscPageSize":"1213",
+                        "dscUnique":timestamp
+                        }
+                }
+        payload = json.dumps({'json':str(body)})
+        req = scrapy.Request(links_url,
+                    method='POST',
+                    body=payload,
+                    headers={
+                        "Content-Type": "application/json",
+                        "Host":"lobw.kultus-bw.de",
+                        "Connection":"keep-alive",
+                        "Accept":"application/json, text/javascript, */*; q=0.01",
+                        "Origin":"https://lobw.kultus-bw.de",
+                        "Referer":"https://lobw.kultus-bw.de/didsuche/"
+                       },
+                    callback=self.parse_schoolist)
+        yield req
+
+    # go on each schools details side
+    def parse_schoolist(self, response):
+        school_data_url = 'https://lobw.kultus-bw.de/didsuche/DienststellenSucheWebService.asmx/GetDienststelle'
+        items = json.loads(json.loads(response.text)['d'])['Rows']
+        for item in items:
+            disch = item['DISCH'][1:-1]  # remove ''
+            payload = json.dumps({'disch':disch})
+            req = scrapy.Request(school_data_url,
+                        method='POST',
+                        body=payload,
+                        headers={
+                            "Content-Type": "application/json",
+                            "Host":"lobw.kultus-bw.de",
+                            "Connection":"keep-alive",
+                            "Accept":"application/json, text/javascript, */*; q=0.01",
+                            "Origin":"https://lobw.kultus-bw.de",
+                            "Referer":"https://lobw.kultus-bw.de/didsuche/"
+                        },
+                        callback=self.parse_school_data)
+            yield req
+
+    # get the information
+    def parse_school_data(self, response):
+        item = json.loads(json.loads(response.body_as_unicode())['d'])
+        data = {
+            'name'             : item['NAME'], 
```

Comment:
I think it could be argued that this should just return the data as is - without remapping the keys. The normalisation could then happen when converting the a `School` item. I'm  fine with merging this as is though

Reply:
+1 to put this into the normalisation step. I think its much cleaner if we do only the scraping in the parsing section of the spider, and everything regarding cleaning the data in our pipeline.

Reply:
@MartinGer would you be ok with making this change? Ideally after #49  is merged. If you do not have time, I'm also happy to take this over.